### ISSUE TITLE
SAH content structuring

### ISSUE THREAD

**Issue by kpettinga** [ObjectiveSubject/cgu#58]
# Summary of to-dos

**Departments**
- [x] Need a way to assign custom templates to Departments. Already doing this elsewhere so should just be a matter of copy/paste/adapt.

_Would like to make the Visiting Artist Lecture series a module. Would need a way to show a carousel of events inside it though. Let's use a shortcode._ ⬇️ ⬇️ 
- [x] Need a shortcode to display 10 most imminent events with the "lectures" `tribe_events_cat` term. Just make it spit out an unordered list of titles for now and I'll build it out later.

**Modules**
- [x] Need a new "history carousel" module type. This module type should have a repeater group with image, year, title, and description fields. We already have an instance of a field group for a history carousel on the home page. See https://github.com/ObjectiveSubject/cgu/blob/master/includes/functions/metaboxes.php#L506-L540. It's probably okay to use default wp media fields for title and description and just create a text field for the year.

**Galleries**
- [x] Create a `gallery` taxonomy on People and Student post types
- [x] Been having trouble with the permalinks for gallery term archives. I created some gallery terms and assigned students to it but I just get redirected to the home page when going to the url (e.g. cgu.local/gallery/mfa-2017-students).
- [x] Would like to have an image field on the gallery term edit page. Found this: https://github.com/jcchavezs/cmb2-taxonomy | but up to you if you want to use it.

**Students**
- [x] Need custom fields on students for the artwork carousel. See #57.

---
🔻 🔻  original comment 🔻 🔻 

---

Hey @davekellam 

I'm building out templates for pages/posts belonging to the School of Arts and Humanities. Would appreciate your input and help making it happen.

Here is the dropbox link to the latest mockups for the School of Arts & Humanities
https://www.dropbox.com/s/o9fkwbl55u913jz/CGU%20Website%20SAH%2020160727.pdf?dl=0

**page 1: SAH landing page**
Pretty straightforward here. This doesn't differ much from the other schools as far as I know. The template's pretty much all done.

**page 2: Program single**
Also pretty straightforward. Nothing unique to this program compared to others. Template is already built.

**page 3: Art Department single**
- Here's where things get interesting. We can create this page using the Department post type although we're probably going to need a special template for the Art Department. All the written content from the intro to the "Featured Programs" section can be entered using the default wysiwyg editor. 
- I figure the "Facilities" block can be created in a module and then assigned to this department post. 
- Next we have an "Art Galleries" carousel. See the note for page 4 for more on this.
- The "Visiting Artist Lecture Series" will have a carousel of events and a list of previous lecturers. Rather than trying to create custom fields for this specific department, it would be great if we could make this block a module that can then just be assigned to the department. The carousel complicates that though. Maybe we could create a shortcode that outputs event carousels like this one?
- "History of CGU Art": again, would be great to create this as a module but it's a bit more complicated that other carousels. This might actually be a case for a new module type. We already have another carousel just like this one on the home page (for which fields are already created). I can imagine they might want to show off the history of other departments as well.

**page 4: Gallery page**
Galleries are just collections of students in an art program ("MFA 2016", for instance). I've been torn between making Art Galleries it's own post type or just a child post of Art Department with a template selection field. A full blown post type seems heavy handed for something that takes up such a small part of the site. But making it a child post of "Art Department" gets kinda strange when I have to build in a template selector. Ultimately I ended up going with the latter, the child-post approach. 
💡💡💡 !!
**Uhh... I just realized that we should make Galleries a taxonomy. 😳  ... duh...
That will allow us to tag students in galleries, and then query those galleries elsewhere.**

If Galleries can be it's own taxonomy, this page is just a taxonomy archive 😄 .

**page 5: Student single**
We can use the student post type here. We'll just need extra custom fields for the artwork carousel. Specs for that are outlined in this issue: #57.
In the galleries below the artwork, we'll just query the "gallery" terms.

**page 6**: (ignore)

---

**So, to summarize:**
- Need a way to assign custom templates to Departments. Already doing this elsewhere so should just be a matter of copy/paste/adapt.
- Need custom fields on students for the artwork carousel. See #57.
- Would like to make the Visiting Artist Lecture series a module. Would need a way to show a carousel of events inside it though (shortcode?).
- We might want to make a new "History Carousel" module type.

**Comment by davekellam** [comment]
* department specific templates should be fine
* can follow up on #57, but we might need to think on galleries a bit more if they can be from faculty or multiple students too
* Module probably make sense to insert lectures (events) on certain pages. Shortcode could work, but it would probably need to use post ids, which could be awkward. Could see if the cmb2 post picker work for events (imagine it should)
* A history module could work... would probably need a repeater in this instance to deal with the image, otherwise we've used shortcodes in the past. Would beed to investigate, image/file fields don't play nicely as repeaters, but might be fine inside a group repeater.

**Comment by kpettinga** [comment]
Hey @davekellam 

- ✔️ 
- ✔️ 
- Yeah we shouldn't use post IDs. The events should be a list of the most imminent (lets say max 10) from a "lectures" term on the `event_category` taxonomy.
- FYI, there's already an instance of a field group for a history carousel on the home page. See https://github.com/ObjectiveSubject/cgu/blob/master/includes/functions/metaboxes.php#L506-L540.

**Comment by kpettinga** [comment]
# Summary of to-dos

**Departments**
- [ ] Need a way to assign custom templates to Departments. Already doing this elsewhere so should just be a matter of copy/paste/adapt.

_Would like to make the Visiting Artist Lecture series a module. Would need a way to show a carousel of events inside it though. Let's use a shortcode._ ⬇️ ⬇️ 
- [ ] Need a shortcode to display 10 most imminent events with the "lectures" `tribe_events_cat` term. Just make it spit out an unordered list of titles for now and I'll build it out later.

**Modules**
- [ ] Need a new "history carousel" module type. This module type should have a repeater group with image, year, title, and description fields. We already have an instance of a field group for a history carousel on the home page. See https://github.com/ObjectiveSubject/cgu/blob/master/includes/functions/metaboxes.php#L506-L540. It's probably okay to use default wp media fields for title and description and just create a text field for the year.

**Galleries**
- [ ] Create a `gallery` taxonomy on People and Student post types
- [ ] Would like to have an image field on the gallery term edit page. Found this: https://github.com/jcchavezs/cmb2-taxonomy | but up to you if you want to use it.

**Students**
- [ ] Need custom fields on students for the artwork carousel. See #57.

---
